e2e/utils.go

```diff
@@ -1273,3 +1273,27 @@ func retryKubectlFile(namespace string, action kubectlAction, filename string, t
 		return true, nil
 	})
 }
+
+// createSecret creates secret given namespace and secret yaml file path.
+func createSecret(c kubernetes.Interface, secretPath, ns string) {
+	secret, err := getSecret(secretPath)
+	if err != nil {
+		e2elog.Failf("failed to load secret: %v", err)
+	}
+	_, err = c.CoreV1().Secrets(ns).Create(context.TODO(), &secret, metav1.CreateOptions{})
```

Comment:
This does not have a retry in case etcdserver is temporary unavailable. Is there a reason you can not use `retryKubectlFile()` for this?